core/src/main/java/io/smallrye/openapi/runtime/io/response/ResponseReader.java

```diff
@@ -179,25 +181,31 @@ public static APIResponse readResponse(final AnnotationScannerContext context,
      */
     public static APIResponse readResponseSchema(final AnnotationScannerContext context,
             final AnnotationInstance annotation) {
-        if (annotation == null || CurrentScannerInfo.getCurrentProduces() == null) {
-            // Only generate the APIResponse if the endpoint declares an @Produces media type
+        if (annotation == null) {
             return null;
         }
         IoLogging.logger.singleAnnotation("@APIResponseSchema");
-        Content content = new ContentImpl();
-
-        for (String mediaType : CurrentScannerInfo.getCurrentProduces()) {
-            MediaType type = new MediaTypeImpl();
-            type.setSchema(SchemaFactory.typeToSchema(context,
-                    JandexUtil.value(annotation, ResponseConstant.PROP_VALUE),
-                    context.getExtensions()));
-            content.addMediaType(mediaType, type);
-        }
 
         APIResponseImpl response = new APIResponseImpl();
         response.setDescription(JandexUtil.value(annotation, ResponseConstant.PROP_RESPONSE_DESCRIPTION));
-        response.setContent(content);
         response.setResponseCode(JandexUtil.value(annotation, ResponseConstant.PROP_RESPONSE_CODE));
+
+        Type responseType = JandexUtil.value(annotation, ResponseConstant.PROP_VALUE);
+
+        if (CurrentScannerInfo.getCurrentProduces() != null && !TypeUtil.isVoid(responseType)) {
+            // Only generate the content if the endpoint declares an @Produces media type
```

Comment:
Checking `void` doesn't really enable anything that couldn't be done with `@APIResponse`, but reduces surprises if `void.class` is given as the `value` for `@APIResponseSchema` and there *is* `@Produces` annotation.